어드민 페이지 입니다.

yu-teacher opened the pull request:
# Pr 종류
- [x] task (개발)
- [ ] bugfix (버그 수정)
- [ ] etc (그 외)

# 무엇이 문제인가요? (What)
어드민 페이지가 없습니다.

# 어떻게 해결하고 구현했나요? (How)
1. 유저 및 사용 디바이스 추가 만들었습니다.
2. 유저 변경창에서 디바이스 추가 가능합니다.
3. 체크박스로 가능하도록 만들었습니다.
4. service계정으로 가입되면 토큰을 db에 저장합니다.
5. 엔트포인트 수정이 조금 일어났습니다.
6. 아직 권한별 기능은 없습니다.

# 테스트 했나요?
- [x] Local Test (본인 pc 에서 테스트)
- [ ] include Test Code (테스트 코드 포함)

yu-teacher committed:
어드민 페이지 입니다.
1. 유저 및 사용 디바이스 추가 만들었습니다.
2. 유저 변경창에서 디바이스 추가 가능합니다.
3. 체크박스로 가능하도록 만들었습니다.
4. service계정으로 가입되면 토큰을 db에 저장합니다.
5. 엔트포인트 수정이 조금 일어났습니다.
6. 아직 권한별 기능은 없습니다.

dodeca05 commented on build.gradle.kts:
```diff
@@ -37,6 +37,12 @@ dependencies {
+	testImplementation("org.projectlombok:lombok:1.18.22")
+	testImplementation("org.projectlombok:lombok:1.18.22")
+	testImplementation("org.projectlombok:lombok:1.18.22")
+	testImplementation("org.projectlombok:lombok:1.18.22")
```
같은 코드가 여러줄인데 정상 맞나요?

dodeca05 commented on src/main/java/com/pro/WOLmgr/controller/AccessController.java:
```diff
@@ -1,35 +1,35 @@
-                new ResponseEntity<>("Access failed.",HttpStatus.UNAUTHORIZED);
+    @GetMapping("/access/{username}")
+    public List<DeviceResponseDTO> readDeviceList(@PathVariable String username){
+        return deviceService.userAccessDevices(username);
```
Rest API 를 구현할려면 /access와 /access/{username}을 둘다 가지고 있어야 합니다.

yu-teacher replied:
용도가 접근가능한 디바이스인데 /access 는 어떤 용도로 만들어야 할지 모르겠습니다.

dodeca05 commented on src/main/java/com/pro/WOLmgr/controller/UserController.java:
```diff
@@ -31,14 +35,29 @@ public class UserController {
+
     @GetMapping("/id/duplication") // 아이디 중복 체크
     public @ResponseBody HashMap<String,String> idCheck(@RequestParam("userId") String userId){
 
```
API의 일괄성을 지켜야 합니다. userName 중복 체크는 path parameter인데 id 중복 체크는 RequestParameter 입니다.
수정하는 김에 Hash도 없앴으면 좋겠습니다.

dodeca05 commented on src/main/java/com/pro/WOLmgr/controller/WebController.java:
```diff
@@ -25,4 +25,10 @@ public void main(){}
     public void addPc(){}
+
+    @GetMapping("/control/index")
+    public void adminIndex(){}
```
만약 이것이 web page와 mapping하는 것이라면 앞에 web 을 붙혀야 됩니다.
그래야 swagger에 나타나지 않습니다.

dodeca05 commented on src/main/java/com/pro/WOLmgr/dto/UserInfoDTO.java:
```diff
@@ -31,4 +31,14 @@ public static UserInfoDTO toDto(UserEntity userEntity) {
+                .email(dto.getEmail())
+                .roles(dto.getRoles())
+                .build();
+    }
```
변환메소드 이름인데 이처럼 개체성을 띌 경우 static을 붙히지 않고 별도의 parameter도 붙히지 않습니다.

yu-teacher replied:
개체성을 띈다는 말씀이 어떤 이야기인지 이해가 가지 않습니다.

dodeca05 commented on src/main/java/com/pro/WOLmgr/dto/UserPrivacyDTO.java:
```diff
@@ -29,6 +32,7 @@ public static UserEntity toEntity(UserPrivacyDTO dto) {
                 .roles(dto.getRoles())
+                .serviceToken(dto.getServiceToken())
                 .build();
     }
```
변환 메소드 이름 네이밍 및 형태가 잘못 되었습니다.
객체 하나를 다른 하나의 객체로 변환할때 static을 쓰지 않고 parameter도 받지 않고 객체 자기 자신이 알아서 변환되도록 해야 됩니다.
이전에도 수정 안된 듯 합니다.
https://github.com/dodeca05/wolMgr/pull/20/files/80653ef544672f3362e11aaf6aa54c750e40d411#r1176419938

dodeca05 commented on src/main/java/com/pro/WOLmgr/service/UserService.java:
```diff
@@ -22,29 +26,49 @@ public class UserService {
-        userDTO.setRoles(roles);
-
-        userRepository.save(toEntity(userDTO));
+        userRepository.save(UserPrivacyDTO.toEntity(userDTO));
```
userDTO.toEntity() 로 되도록 수정합시다.

yu-teacher replied:
이 부분에 어느 부분이 잘못 되었을까요?

dodeca05 commented on src/main/resources/static/js/isEmail.js:
```diff
@@ -55,7 +55,7 @@ function handleSignUp() {
 
     // 회원가입에 필요한 데이터 생성
-    let data = { userId, password, username, email }
+    let data = { userId, password, username, email, "roles":[USER] }
```
유저 권한을 회원 가입할때 받는 행위는 보안적으로 매우 위험한 행위 입니다.

dodeca05 replied:
role부분에 admin이 들어가면 생성이 안되는지 테스트가 필요합니다.

yu-teacher committed:
USER 가입 관련 제외 수정 했습니다.
1. to, from 관련 수정 했습니다.
2. access 엔드포인트 수정 했습니다.
3. 그래들 수정 했습니다.
4. 프론트 매핑 변경 했습니다.

yu-teacher committed:
USER 가입 관련 제외 수정 했습니다.
1. to, from 관련 수정 했습니다.
2. access 엔드포인트 수정 했습니다.
3. 그래들 수정 했습니다.
4. 프론트 매핑 변경 했습니다.
5. 회원가입 로직을 백단으로 옮겼습니다.

yu-teacher committed:
USER 가입 관련 제외 수정 했습니다.
1. to, from 관련 수정 했습니다.
2. access 엔드포인트 수정 했습니다.
3. 그래들 수정 했습니다.
4. 프론트 매핑 변경 했습니다.
5. 회원가입 로직을 백단으로 옮겼습니다.
6. 토큰 분리 하였습니다.
7. 헤더에 유효한 토큰이 있다면 여려 권한을 줄 수 있고 만약 유효한 토큰이 없거나 헤더가 없다면 무조건 USER가 됩니다.

yu-teacher commented:
USER 가입 관련 제외 수정 했습니다.
1. to, from 관련 수정 했습니다.
2. access 엔드포인트 수정 했습니다.
3. 그래들 수정 했습니다.
4. 프론트 매핑 변경 했습니다.
5. 회원가입 로직을 백단으로 옮겼습니다.
6. 토큰 분리 하였습니다. 
7. 헤더에 유효한 토큰이 있다면 여려 권한을 줄 수 있고 만약 유효한 토큰이 없거나 헤더가 없다면 무조건 USER가 됩니다.

dodeca05 reviewed (approved):
컴플릭을 조심해서 처리하시길 바랍니다.

yu-teacher committed:
Merge remote-tracking branch 'origin/main' into task/15/adminPageDemo

# Conflicts:
#	src/main/java/com/pro/WOLmgr/Jwt/JwtAuthorizationFilter.java
#	src/main/java/com/pro/WOLmgr/controller/UserController.java
#	src/main/resources/static/js/commonAuth.js
